pkg/cluster/sync.go

```diff
@@ -303,6 +312,42 @@ func (c *Cluster) syncStatefulSet() error {
 	return nil
 }
 
+// checkAndSetGlobalPostgreSQLConfiguration checks whether cluster-wide API parameters
+// (like max_connections) has changed and if necessary sets it via the Patroni API
+func (c *Cluster) checkAndSetGlobalPostgreSQLConfiguration() error {
+	// we need to extract those options from the cluster manifest.
+	optionsToSet := make(map[string]string)
+	pgOptions := c.Spec.Parameters
+
+	for k, v := range pgOptions {
+		if isBootstrapOnlyParameter(k) {
+			optionsToSet[k] = v
+		}
+	}
+
+	if len(optionsToSet) > 0 {
+		pods, err := c.listPods()
+		if err != nil {
+			return err
+		}
+		if len(pods) == 0 {
+			return fmt.Errorf("could not call Patroni API: cluster has no pods")
+		}
+		for _, pod := range pods {
+			podName := util.NameFromMeta(pod.ObjectMeta)
+			c.logger.Debugf("calling Patroni API on a pod %s to set the following Postgres options: %v",
+				podName, optionsToSet)
+			if err := c.patroni.SetPostgresParameters(&pod, optionsToSet); err == nil {
+				return nil
+			} else {
+				c.logger.Warningf("could not patch postgres parameters with a pod %s: %v", podName, err)
+			}
+		}
+		return fmt.Errorf("could not call Patroni API to set Postgres options: failed on %d pods", len(pods))
```

Comment:
What exactly does this error message intend to say ? That if Patroni fails to apply a param, it fails on all pods at once ?

Reply:
That calling Patroni API failed, and we tried %d pods.

Reply:
> we tried %d pods.

i first understood the original err msg as "update failed on all `len(pods)` pods"